### ISSUE TITLE
About the ResponseTimeoutException

### ISSUE THREAD

**Issue by SeaCN** [opentelecoms-org/jsmpp#82]
Dear:
     

> No response after waiting for 2000 millis when executing unbind with sessionId cd173506 and sequenceNumber 3.
this is the exception when session.unbindAndClose().  Plase tell how to resolve this problem ,I didn't find how to set the waiting for unbindAndClose()

**Comment by SeaCN** [comment]
I have set the transmittime to 5000,but still the same exception.Do you have any idea for this issue,any possible reason?

**Comment by pmoerenhout** [comment]
DO you have a tcpdump of the disconnect? It could be that the server sends the unbind response, and immediately disconnects. In that case, the unbind response could be lost at the server as the sending was not finished yet.

**Comment by SeaCN** [comment]
Does the server mean smsc?

**Comment by pmoerenhout** [comment]
Yes, when the SMSC receives the unbind request, it must send the unbind response, but will also close the connection. Normally it has to wait a small time to let the unbind response be transmitted, before closing the TCP connection.

**Comment by SeaCN** [comment]
If I do session.unbindandclose method immediately after submitsm,can I receiver the deliverysm response?

**Comment by pmoerenhout** [comment]
No. Normally the SMPP connection is a fairly permanent connection. If you need to get deliversm's, you need to have a bounded connection to the SMSC (via SMPP).
Normally you bind, and keep that connection open. If the enquire fails, the connection is dropped, or other error, you can drop the connection and start a new one.
You can also have multiple TCP connections towards the same SMSC. You can also use connection pooling. I have some examples of that.
If you unbound, the SMSC cannot deliver. There is also a mechanism whereby the SMSC build the TCP connection to the ESME, but this is not often used.

**Comment by SeaCN** [comment]
Thank you very much， this help me a lot!

**Comment by SeaCN** [comment]
@pmoerenhout you said you have some examples of connection pooling, can you give me them? Thank you

**Comment by SeaCN** [comment]
Dear @pmoerenhout , please give me back when seeing this message. I really need those examles of connection pooling. Thank you

**Comment by pmoerenhout** [comment]
I have added the example for the connection pooling to my GitHub: https://github.com/pmoerenhout/jsmpp-pool
